lib/pbench/test/unit/server/query_apis/test_controllers_list.py

```diff
@@ -91,33 +53,15 @@ def test_missing_keys(
         Note that "start", and "end" are required whereas "user" is not mandatory;
         however, Pbench will silently ignore any additional keys that are
         specified.
-        """
-        response = client.post(
-            f"{server_config.rest_uri}/controllers/list",
-            headers={"Authorization": "Bearer " + pbench_token},
-            json=keys,
-        )
-        assert response.status_code == HTTPStatus.BAD_REQUEST
-        missing = [k for k in ("start", "end") if k not in keys]
-        assert (
-            response.json.get("message")
-            == f"Missing required parameters: {','.join(missing)}"
-        )
-
-    def test_bad_dates(self, client, server_config, user_ok, pbench_token):
-        """
-        Test behavior when a bad date string is given
-        """
-        response = client.post(
-            f"{server_config.rest_uri}/controllers/list",
-            headers={"Authorization": "Bearer " + pbench_token},
-            json={"user": "drb", "start": "2020-12", "end": "2020-19"},
-        )
-        assert response.status_code == HTTPStatus.BAD_REQUEST
-        assert (
-            response.json.get("message")
-            == "Value '2020-19' (str) cannot be parsed as a date/time string"
-        )
+       """
+        self.required_keys = [
+            key
+            for key, parameter in ControllersList(
+                client.config, client.logger
+            ).schema.parameters.items()
```

Comment:
An _instance_ of the class?...hmm...yes...could do...but...this is Python, where we can pass the class itself, and use that object to create an instance when we want it!  😀 

I dunno, maybe it would be better to use an instance...I just get so few opportunities to use class objects.  😉

Reply:
The code excerpt above instantiates an instance of `ControllersList` and then invokes a method on one of its attributes; instead, it could do the instantiation here and let the invocation be done elsewhere..._or_...it could pass the class object so that the code elsewhere could do _both_ the instantiation and the invocation.  (It's pretty close to six of one and a half dozen of the other, I think....)